jcl/src/java.base/share/classes/openj9/internal/tools/attach/target/DiagnosticUtils.java

```diff
@@ -348,6 +387,128 @@ private static DiagnosticProperties doDump(String diagnosticCommand) {
 		return result;
 	}
 
+/*[IF JFR_SUPPORT]*/
+	private static Long convertToMilliseconds(String timeValue) {
+		long timeInMilli = 0L;
+
+		// extract the numeric part and the time unit
+		String numericPart = timeValue.replaceAll("[^0-9]", "");
+
+		// convert the unit to lowercase for consistency
+		String timeUnit = timeValue.replaceAll("[0-9]", "").toLowerCase();
+
+		// parse the numeric part
+		long time = Long.parseLong(numericPart);
+
+		// convert to milliseconds based on the unit
+		switch (timeUnit) {
+		case "s":
+			timeInMilli = TimeUnit.SECONDS.toMillis(time);
+			break;
+		case "m":
+			timeInMilli = TimeUnit.MINUTES.toMillis(time);
+			break;
+		case "h":
+			timeInMilli = TimeUnit.HOURS.toMillis(time);
+			break;
+		case "d":
+			timeInMilli = TimeUnit.DAYS.toMillis(time);
+			break;
+		default:
+			// no unit or unrecognized unit, assume milliseconds
+			timeInMilli = time;
+			break;
+		}
+		return Long.valueOf(timeInMilli);
+	}
+
+	private static Long parseTimeParameter(String paramName, String[] parameters) {
+		for (String param : parameters) {
+			if (param.startsWith(paramName + "=")) {
+				int valueStart = param.indexOf("=");
+				if (valueStart != -1) {
+					return convertToMilliseconds(param.substring(valueStart + 1));
+				}
+			}
+		}
+		return null;
+	}
+
+	private static String parseStringParameter(String paramName, String[] parameters, String defaultValue) {
+		for (String param : parameters) {
+			if (param.startsWith(paramName + "=")) {
+				int valueStart = param.indexOf("=");
+				if (valueStart != -1) {
+					return param.substring(valueStart + 1);
+				}
+			}
+		}
+		return defaultValue;
+	}
+
+	private static DiagnosticProperties doJFR(String diagnosticCommand) {
+		DiagnosticProperties result = null;
+
+		// split the command and arguments
+		String[] parts = diagnosticCommand.split(DIAGNOSTICS_OPTION_SEPARATOR);
+		IPC.logMessage("doJFR: ", diagnosticCommand);
+
+		// ensure there's at least one part for the command
+		if (parts.length == 0) {
+			return DiagnosticProperties.makeErrorProperties("Error: No JFR command specified");
+		}
+
+		String command = parts[0].trim();
+		String[] parameters = Arrays.copyOfRange(parts, 1, parts.length);
+
+		String fileName = parseStringParameter("filename", parameters, null);
+		IPC.logMessage("doJFR: filename = ", fileName);
+		boolean setFileName = (fileName != null) && !fileName.isEmpty();
+		if (setFileName) {
+			if (!VM.setJFRRecordingFileName(fileName)) {
+				return DiagnosticProperties.makeErrorProperties("setJFRRecordingFileName failed");
+			} else {
+				jfrRecordingFileName = fileName;
+			}
+		}
+
+		try {
+			if (command.equalsIgnoreCase(DIAGNOSTICS_JFR_START)) {
+				VM.startJFR();
```

Comment:
we should only allow one recording at a time

Reply:
Second and later `JFR.start()` prints an error message `One JFR recording is in progress [jfrRecordingFileName], only one recording is allowed at a time.`